### ISSUE TITLE
hbomax.com

### ISSUE THREAD

**Issue by asd3210** [yt-dlp/yt-dlp#3145]
### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.03.08.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

Czech Republic

### Example URLs

C:\DL>yt-dlp [https://play.hbomax.com/episode/urn:hbo ... yZIQEAAABq](https://play.hbomax.com/episode/urn:hbo:episode:GYTf0SgK-jQyZIQEAAABq)
[generic] urn:hbo:episode:GYTf0SgK-jQyZIQEAAABq: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] urn:hbo:episode:GYTf0SgK-jQyZIQEAAABq: Downloading webpage
[generic] urn:hbo:episode:GYTf0SgK-jQyZIQEAAABq: Extracting information
ERROR: Unsupported URL: [https://play.hbomax.com/episode/urn:hbo ... yZIQEAAABq](https://play.hbomax.com/episode/urn:hbo:episode:GYTf0SgK-jQyZIQEAAABq)

### Description

Dears,

could you please add this hbomax.com if its possible. to yt-dlp support sites?

Thank you very much

Kind Regards,

### Verbose log

```shell
C:\DL>yt-dlp -vU
[debug] Command-line config: ['-vU']
[debug] Encodings: locale cp1250, fs utf-8, out utf-8, err utf-8, pref cp1250
[debug] yt-dlp version 2022.03.08.1 [c0c2c57] (win_exe)
[debug] Python version 3.8.10 (CPython 64bit) - Windows-10-10.0.19044-SP0
[debug] exe versions: none
[debug] Optional libraries: brotli, Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
Latest version: 2022.03.08.1, Current version: 2022.03.08.1
yt-dlp is up to date (2022.03.08.1)
```

**Comment by zackmark29** [comment]
That's not possible. It's DRM Protected only

**Comment by pukkandan** [comment]
I can't verify this coz of geoblock, but I'll trust you know what u r talking about

**Comment by Vangelis66** [comment]
> I can't verify this coz of geoblock, but I'll trust you know what u r talking about

I used a whitelisted **US IP address** and can verify **wxolp**'s claim! :+1: 
Browser used was an **EME-free** (no WidevineCDM support) **Chromium** build, I had no issues starting the stream at 
```
https://www.hbomax.com/collections/watch-free/barry-s1-e1
```
even **without an HBOmax account**!

The [Stream Detector](https://github.com/rowrawer/stream-detector/tags) open-source extension had no problem "detecting" the **DRM-free** (IOW, **clear**) HLS master playlist below: 
```
https://clips-media-lln.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/master_cl_de.m3u8
```
which, as an added bonus, **isn't geo-fenced** itself: 
```
yt-dlp -F "https://clips-media-lln.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/master_cl_de.m3u8"

[generic] master_cl_de: Downloading webpage
[generic] master_cl_de: Downloading m3u8 information
[info] Available formats for master_cl_de:
ID            EXT RESOLUTION |   TBR PROTO | VCODEC        VBR ACODEC     MORE INFO
--------------------------------------------------------------------------------------
audio-ENGLISH mp4 audio only |       m3u8  | audio only        unknown    [en] ENGLISH
334           mp4 416x234    |  335k m3u8  | avc1.4d400d  335k video only
634           mp4 416x234    |  635k m3u8  | avc1.4d400d  635k video only
1134          mp4 640x360    | 1135k m3u8  | avc1.4d401e 1135k video only
1935          mp4 768x432    | 1935k m3u8  | avc1.4d401f 1935k video only
2635          mp4 960x540    | 2636k m3u8  | avc1.4d401f 2636k video only
5137          mp4 1280x720   | 5137k m3u8  | avc1.4d401f 5137k video only
8138          mp4 1920x1080  | 8138k m3u8  | avc1.640028 8138k video only
```

**Comment by wxolp** [comment]
> **Later Edit**: HBOmax first check your geo-location and set 3 location-related **browser cookies** ("countryCode", "city", "continent"); these 3 have a lifespan of 11hr; when these cookies are first set from a non-sanctioned location, using a proxy/VPN to spoof a US location won't work in the browser, unless those 3 cookies are first eliminated and page reloaded...

@Vangelis66 good catch. I have seen other sites like Stack Exchange use similar "poison cookies". Dirty tactic.

**Comment by pukkandan** [comment]
I always clear cookies when testing for geo-block. Sadly it seems my VPN is blacklisted

**Comment by bashonly** [comment]
won't be supported
